src/components/PrimaryNav/PrimaryNav.tsx

```diff
@@ -73,10 +73,18 @@ const ControllersLink = () => {
 const PrimaryNav = () => {
   const appVersion = useSelector(getAppVersion);
   const [updateAvailable, setUpdateAvailable] = useState(false);
+  const [versionRequested, setVersionRequested] = useState(false);
 
-  dashboardUpdateAvailable(appVersion || "").then((e) => {
```

Comment:
I missed this in review, but this call was causing an infinite loop as it would be called on every render, and after each API response it would update state, causing a new render which started the process again.

Reply:
Sorry about that!

Reply:
Also thanks for the mock! for some reason `jest.mock` was just not working in this scenario.